addons/crm/models/crm_lead.py

```diff
@@ -1466,20 +1479,60 @@ def _cron_update_automated_probabilities(self):
             pending_lead_domain = ['&', '&', ('stage_id', '!=', False), ('create_date', '>', pls_start_date),
                                    '|', ('probability', '=', False), '&', ('probability', '<', 100), ('probability', '>', 0)]
             leads_to_update = self.env['crm.lead'].search(pending_lead_domain)
-            lead_probabilities = leads_to_update._pls_get_naive_bayes_probabilities(batch_mode=True)
+            nbr_leads_to_update = len(leads_to_update)
+
+            # Compute by batch to avoid memory error
+            batch_step = 500000
+            lead_probabilities = {}
+            for i in range(0, nbr_leads_to_update, batch_step):
+                leads_to_update_part = leads_to_update[i:i+batch_step]
```

Comment:
Zizisse the part where I wonder if prefetching will not lead to reading everything from db (cfr comment hereabove).